game_patch/hud/multi_scoreboard.cpp

```diff
@@ -53,14 +53,17 @@ int draw_scoreboard_header(int x, int y, int w, rf::NetGameType game_type, bool
 
     // Draw Game Type name
     if (!dry_run) {
-        const char* game_type_name;
-        if (game_type == rf::NG_TYPE_DM)
-            game_type_name = rf::strings::deathmatch;
-        else if (game_type == rf::NG_TYPE_CTF)
-            game_type_name = rf::strings::capture_the_flag;
-        else
-            game_type_name = rf::strings::team_deathmatch;
-        rf::gr::string_aligned(rf::gr::ALIGN_CENTER, x_center, cur_y, game_type_name);
+        int num_players = rf::multi_num_players();
+        std::string player_count_str =
```

Comment:
could number of heap allocations be reduced? `game_type_name` don't have to be an `std::string`, it could be `const char*` or `std::string_view`. The string used for concatenation could have reserved memory before concatenation.
This code runs every frame so it should be fast and allocations are generally considered slow and can lead to micro-stutters. Also please consider using `std::format_to`